src/sentry/integrations/slack/utils/channel.py

```diff
@@ -122,14 +130,16 @@ def get_channel_id_with_timeout(
     }
 
     time_to_quit = time.time() + timeout
-
-    client = SlackClient(integration_id=integration.id)
+    if has_sdk_flag:
+        client = SlackSdkClient(integration_id=integration.id)
+    else:
+        client = SlackClient(integration_id=integration.id)
     id_data: tuple[str, str | None, bool] | None = None
     found_duplicate = False
     prefix = ""
     channel_id = None
     try:  # Check for channel
-        channel_id = check_for_channel(client, name)
+        channel_id = check_for_channel(client, name, has_sdk_flag)
```

Comment:
do we need to be passing the sdk flag if we can infer which method to use (`client.post` or `client.chat_postMessage`) based on the type of the client?